### DISCUSSION TITLE
Add a return_adv option to all find_device_by_* methods in BleakScanner

### DISCUSSION THREAD

**Discussion by ekenyon676** [hbldh/bleak#1597] [Ideas]
`BleakScanner` has the option to return advertisement data in its `discover` method, but only returns the device in the `find_device_by_address`, `find_device_by_name`, and `find_device_by_filter` functions. It can often be useful to return the advertisement data in those other situations as well. I've basically had to re-implement those functions in my application because I want both the device and ad data for a device I'm looking for by name.

**Comment by dlech** [comment]
See https://github.com/hbldh/bleak/issues/1277

**Reply by ekenyon676** [reply]
Thanks! I searched `return_adv`, but I guess not in the issues section. Is this something you're willing to take a PR for? I don't see "help wanted" on the issue.

**Reply by dlech** [reply]
Yes, would be happy to take a PR (every issue is "help wanted" :smile:).

Instead of modifying the existing methods to return a different value based on the args, I think I would rather just add a new methods like `find_device_and_data_by_address(...) -> Tuple[...]`. I think this will work better with static analysis and code completion tools.

(And the existing functions could be modified to call the new functions internally and discard the advertising data to avoid duplicating code)

**Reply by ekenyon676** [reply]
Interesting, I was thinking an input to keep it consistent with the discover API, but I can see that "device" has a specific meaning that is not really covered by discover. Happy to try my hand a PR sometime!
